```
models/workflow.py:339: error: Incompatible return value type (got "bool | None", expected "bool")  [return-value]
services/dataset_service.py:658: error: Incompatible return value type (got "bool | None", expected "bool")  [return-value]
services/workflow_service.py:90: error: Incompatible return value type (got "bool | None", expected "bool")  [return-value]
```

--- a/api/models/workflow.py
+++ b/api/models/workflow.py
@@ -336,7 +336,6 @@ class Workflow(Base):
         """
         from models.tools import WorkflowToolProvider
 
-
         stmt = select(
             exists().where(
                 WorkflowToolProvider.tenant_id == self.tenant_id,

--- a/api/models/workflow.py
+++ b/api/models/workflow.py
@@ -923,7 +923,7 @@ def _naive_utc_datetime():
 
 class WorkflowDraftVariable(Base):
     """`WorkflowDraftVariable` record variables and outputs generated during
-    debugging worfklow or chatflow.
+    debugging workflow or chatflow.
 
     IMPORTANT: This model maintains multiple invariant rules that must be preserved.
     Do not instantiate this class directly with the constructor.

--- a/api/models/workflow.py
+++ b/api/models/workflow.py
@@ -336,14 +336,14 @@ class Workflow(Base):
         """
         from models.tools import WorkflowToolProvider
 
-        return db.session.scalar(
-            select(
-                exists().where(
-                    WorkflowToolProvider.tenant_id == self.tenant_id,
-                    WorkflowToolProvider.app_id == self.app_id,
-                )
+
+        stmt = select(
+            exists().where(
+                WorkflowToolProvider.tenant_id == self.tenant_id,
+                WorkflowToolProvider.app_id == self.app_id,
             )
         )
+        return db.session.execute(stmt).scalar_one()
 
     @property
     def environment_variables(self) -> Sequence[StringVariable | IntegerVariable | FloatVariable | SecretVariable]:

--- a/api/services/dataset_service.py
+++ b/api/services/dataset_service.py
@@ -655,7 +655,8 @@ class DatasetService:
 
     @staticmethod
     def dataset_use_check(dataset_id) -> bool:
-        return db.session.scalar(select(exists().where(AppDatasetJoin.dataset_id == dataset_id)))
+        stmt = select(exists().where(AppDatasetJoin.dataset_id == dataset_id))
+        return db.session.execute(stmt).scalar_one()
 
     @staticmethod
     def check_dataset_permission(dataset, user):

--- a/api/services/workflow_service.py
+++ b/api/services/workflow_service.py
@@ -87,15 +87,14 @@ class WorkflowService:
         )
 
     def is_workflow_exist(self, app_model: App) -> bool:
-        return db.session.scalar(
-            select(
-                exists().where(
-                    Workflow.tenant_id == app_model.tenant_id,
-                    Workflow.app_id == app_model.id,
-                    Workflow.version == Workflow.VERSION_DRAFT,
-                )
+        stmt = select(
+            exists().where(
+                Workflow.tenant_id == app_model.tenant_id,
+                Workflow.app_id == app_model.id,
+                Workflow.version == Workflow.VERSION_DRAFT,
             )
         )
+        return db.session.execute(stmt).scalar_one()
 
     def get_draft_workflow(self, app_model: App) -> Optional[Workflow]:
         """

